project/src/components/card-list/card-list.tsx

```diff
@@ -1,9 +1,33 @@
 import Card from '../card/card';
+import {Offer} from '../../types/offer';
+import {useState} from 'react';
+
+type CardListProps = {
+  offers: Offer[];
+}
+
+function CardList ({offers}: CardListProps): JSX.Element {
+  const [offersId, setOffersId] = useState(0);
+
+  const boxMouseEnterHandler = (newId: number) => {
+    setOffersId((prevId) => {
+      prevId = newId;
+      return prevId;
+    })
```

Comment:
Странная конструкция, зачем чем записывать новый айди как предыдущий и возвращать его?
```suggestion
    setOffersId(newId)
```